### PR TITLE
Hide job listings management link when smaller device

### PR DESCRIPTION
**Pull request by miyabitanimchi**
What I have done on this branch is as follows:

- Hide "求人投稿・管理"( `/joblistings_management`) link when it is on a smaller device (tablet and mobile)
- Created page not found page and enable it to be shown if a user visit the `/joblistings_management` page via the url, and also shown on any other url which don't exist.
- Created data of the pathnames and the function `judgeWrapperLayout()` to judge how the layout should be like and implemented it on the wrapper component
- Refactored code on some components

Please check when you have time, thank you!

### CODE REVIEW

File: src/router/judgeWrapperLayout.js [hunk 1]
```diff
@@ -1,3 +1,59 @@
...
+
+  const isExactMatchHeader = pathnamesExactMatchHeader.includes(currentPath);
+  let isPartialMatchHeader = false;
+  pathnamesPartialMatchHeader.forEach((pathname) => {
+    if (currentPath.includes(pathname)) {
+      isPartialMatchHeader = true;
+    }
+  });
+
+  layout["header"] = isExactMatchHeader || isPartialMatchHeader;
```
Comment (BlackStone8960) [inline src/router/judgeWrapperLayout.js:34]: Can you make this process to function to shorten whole process?

**Review by BlackStone8960 — COMMENTED** [review]
Thank you so much, it look great. 
But can you just fix the point I pointed out in the comment?

Thank you!

**Comment by miyabitanimchi** [comment]
I refactored code you mentioned [above](https://github.com/sh1mot4ku/GDS/pull/43#discussion_r767553473).
Please review it again when you can, thank you 🙏

File: src/router/judgeWrapperLayout.js [hunk 2]
```diff
@@ -1,3 +1,53 @@
...
+
+  const isPathnamePartialMatch = (pathnames) => {
+    let isPathPartialMatch = false;
+    pathnames.forEach((pathname) => {
+      if (currentPath.includes(pathname)) {
+        isPathPartialMatch = true;
+      }
+    });
+    return isPathPartialMatch;
+  };
```
Comment (BlackStone8960) [inline src/router/judgeWrapperLayout.js:34]: Can you declare this function outside of judgeWrapperLayout?

**Comment by miyabitanimchi** [comment]
I put `isPathnamePartialMatch()` function outside `judgeWrapperLayout()` function. Please check again when you can.

**Review by BlackStone8960 — APPROVED** [review]
Thank you very much for doing such a great job! It looks good.

**Pull request merged**